casper/src/main/resources/AsyncMap4.rho

```diff
@@ -0,0 +1,213 @@
+// Rholang map desiderata: speedy insert & lookup,  no conflicts on lookup, no conflicts on inserts to different keys
+// This implementation: O(log n) insert & lookup.
+// Conflict analysis
+//   Lookup
+//     The cell implementation uses a linear for, and when two processes send
+//     to the same linear for, you get a conflict. When looking up an item when
+//     you don't know if it's in the map or not, it necessarily sends "get" to
+//     the root cell of the map, so lookups of that kind conflict.  When
+//     looking up an item where you know it's in the map, you can use
+//     "fastUnsafeGet" and avoid the root cell.
+//   Insert
+//     As above, inserting necessarily sends a "get" to the root cell, thus
+//     inserts conflict.
+
+new AsyncMap, o(`rho:io:stdout`) in {
+  new MakeCell, ByteArrayToNybbleList, AsyncMapSetter, AsyncMapGetter, powersCh in {
+    powersCh!([1,2,4,8,16,32,64,128,256,512,1024,2048,4096,8192,16384,32768, 65536]) |
+    for (@powers <- powersCh) {
+      contract MakeCell(@val, cell) = {
+        for (@cmd, @args <- cell) {
+          match cmd {
+            "get" => {
+              match args {
```

Comment:
you could save one match by doing:

```
        for (@args <- cell) {
          match args {
            [@"get", ret] => {
```
I'd also use a tuple for that purpose

Reply:
MakeCell rewritten entirely.